src/message/MoveMessage.js

```diff
@@ -0,0 +1,211 @@
+/* @flow strict-local */
+import React, { useState, useContext } from 'react';
+import { Text, View, Platform, Picker, TouchableOpacity, ScrollView } from 'react-native';
+import type { Node } from 'react';
+import { ThemeContext, BRAND_COLOR } from '../styles';
+import type { RouteProp } from '../react-navigation';
+import * as api from '../api';
+import Input from '../common/Input';
+import { streamNarrow, streamIdOfNarrow } from '../utils/narrow';
+import { getStreamForId } from '../subscriptions/subscriptionSelectors';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import { getAuth, getStreams, getOwnUser } from '../selectors';
+import { useSelector } from '../react-redux';
+import { showErrorAlert, showToast } from '../utils/info';
+import { Icon } from '../common/Icons';
+import type { Narrow, Message, Outbox } from '../types';
+import TopicAutocomplete from '../autocomplete/TopicAutocomplete';
+import { TranslationContext } from '../boot/TranslationProvider';
+import ZulipButton from '../common/ZulipButton';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'move-message'>,
+  route: RouteProp<'move-message', {| message: Message | Outbox, messageNarrow: Narrow |}>,
```

Comment:
What's the intended meaning of `messageNarrow`? This could use some jsdoc here to clarify.

I'm not sure what it's supposed to mean here. That in turn is necessary in order to properly review the logic below that consumes it.